src/matcha_ml/services/terraform_service.py

```diff
@@ -21,7 +21,7 @@ class TerraformConfig:
     @property
     def state_file(self) -> str:
         """The path to the state file."""
-        return os.path.join(self.working_dir, "matcha.state")
+        return os.path.join(self.working_dir, "..", "matcha.state")
```

Comment:
`..` can be repalced by `os..path.pardir`. 

> `os.path.pardir` stands for ".." which means one directory above the current one.

Also, please run everything after making this change to be sure nothing is breaking and `matcha.state` file is created in correct place.